apps/ui-library/registry/default/examples/infinite-query-hook-demo.tsx

```diff
@@ -0,0 +1,147 @@
+'use client'
+
+import { cn } from '@/lib/utils'
+import {
+  SupabaseQueryHandler,
+  SupabaseTableData,
+  SupabaseTableName,
+  useInfiniteQuery,
+} from '@/registry/default/blocks/infinite-query-hook/hooks/use-infinite-query'
+import { Database } from '@/registry/default/fixtures/database.types'
+import * as React from 'react'
+
+interface InfiniteListProps<TableName extends SupabaseTableName> {
+  tableName: TableName
+  columns?: string
+  pageSize?: number
+  trailingQuery?: SupabaseQueryHandler<TableName>
+  renderItem: (item: SupabaseTableData<TableName>, index: number) => React.ReactNode
+  className?: string
+  renderNoResults?: () => React.ReactNode
+  renderEndMessage?: () => React.ReactNode
+  renderSkeleton?: (count: number) => React.ReactNode
+}
+
+const DefaultNoResults = () => (
+  <div className="text-center text-muted-foreground py-10">No results.</div>
+)
+
+const DefaultEndMessage = () => (
+  <div className="text-center text-muted-foreground py-4 text-sm">You&apos;ve reached the end.</div>
+)
+
+const defaultSkeleton = (count: number) => (
+  <div className="flex flex-col gap-2 px-4">
+    {Array.from({ length: count }).map((_, index) => (
+      <div key={index} className="h-4 w-full bg-muted animate-pulse" />
+    ))}
+  </div>
+)
+
+export function InfiniteList<TableName extends SupabaseTableName>({
+  tableName,
```

Comment:
needs a default? 
tableName = 'products' as TableName,

Reply:
No, this is a component that you can copy into your code to use it.